### DISCUSSION TITLE
Add *.3dm (rhino) file importing

### DISCUSSION THREAD

**Discussion by normanrichardson** [robbievanleeuwen/section-properties#73] [Ideas]
I've been working on a project that converts [rhino geometry to shapely geometry](https://github.com/normanrichardson/rhino_shapely_interop) using [rhino3dm](https://pypi.org/project/rhino3dm/). Its gone fairly well (I think) and its at a place where it could be distributed (I think). 

Is there any interest in adding this functionality? Specifically for #66 (any comments @connorferster, @Czarified, @Agent6-6-6). It would be similar to what is being worked on for [dxf files](https://github.com/connorferster/section-properties/blob/shapely_preprocessor/sectionproperties/pre/dxf.py). A comparable example is [here](https://github.com/normanrichardson/rhino_shapely_interop/blob/master/example1.py).

I am happy to add this functionality to #66, following what is in dxf.py.

**Comment by connorferster** [comment]
I think thats a great idea. Perhaps do a fork of the repo and create a new branch. Add the new functionality as a separate module (like dxf.py, except rhino.py? Or 3dm.py?) and then write some pytest-style tests for it in a new module in the tests directory.

I have been teaching a Python course for the last several weeks. Hope to jump back on to sectionproperties in the first couple of weeks in August.

**Reply by normanrichardson** [reply]
Ok great I'll work on this. Should have it done in the week of the 25-30. 

I browsed a little more and I found that the implementation for dxf is [here](https://github.com/connorferster/section-properties/blob/b18c96f3cd4bd238d57d4b7308c3ef5d090d391d/sectionproperties/pre/sections.py#L198) and [here](https://github.com/connorferster/section-properties/blob/b18c96f3cd4bd238d57d4b7308c3ef5d090d391d/sectionproperties/pre/sections.py#L1368) inside section.py.

To clarify, if we want to put the functionality under Geometry (as something like Geometry.load_3dm(...)) then making it a separate module (say rhino.py) may not work as we'll get a circular import (as rhino.py will try import Geometry to use the Geometry and CompoundGeometry classes). Does this make sense? Or is there something I'm missing?

**Comment by connorferster** [comment]
The idea would be to write the “rhino.py” module as a series of pure functions that takes a file path as input and returns a shapely geometry. 

Then from pre.pre.sections, I can import rhino.py and write a small class method that takes a file path argument and returns a Geometry, using the returned Shapley polygon as the input polygon for the Geometry class.

Let me know if that makes sense.

**Reply by normanrichardson** [reply]
Great, this makes sense: the functions in rhino.py will return shapely geometry and not sectionproperties geometry. I will begin implementing this today or tomorrow, and I will let you know if I need further clarification.

**Comment by robbievanleeuwen** [comment]
+1 to this idea!
